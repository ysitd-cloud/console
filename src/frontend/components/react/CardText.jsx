import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export default function CardText(props) {
  const { className, children, ...others } = props;
  const classes = classNames(className, 'card__text');
  return <div className={classes} {...others}>{children}</div>;
}

CardText.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
};

CardText.defaultProps = {
  className: '',
  children: null,
};
