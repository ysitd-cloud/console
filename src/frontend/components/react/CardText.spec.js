import React from 'react';
import renderer from 'react-test-renderer';
import CardText from './CardText';

/* globals describe, expect, it */

describe('React CardText Component', () => {
  it('match snapshot', () => {
    const component = renderer.create(<CardText className="xs12" id="test">foo</CardText>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
