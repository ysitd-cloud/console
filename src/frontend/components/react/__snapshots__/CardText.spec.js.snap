// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React CardText Component match snapshot 1`] = `
<div
  className="xs12 card__text"
  id="test"
>
  foo
</div>
`;
